harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -221,7 +221,7 @@ def __init__(
         else:
             self.trial_id = self.core_context.train._trial_id
 
-        self.state = _TrialState(trial_id=self.trial_id)
+        self.state = None  # type: Optional[_TrialState]
```

Comment:
might be worth a comment explaining the purpose of the late-initialization, since this could be construed as opposite of what we normally prefer.